client/src/templates/Challenges/redux/code-storage-epic.js

```diff
@@ -136,9 +137,18 @@ function loadCodeEpic(action$, state$) {
       const fileKeys = challengeFiles.map(x => x.fileKey);
       const invalidForLegacy = fileKeys.length > 1;
       const { title: legacyKey } = challenge;
-
       const codeFound = getCode(id);
-      if (codeFound && isFilesAllPoly(codeFound)) {
+
+      // see if this challenge is saved in database
+      const savedChallenges = savedChallengesSelector(state);
+      const savedChallenge = savedChallenges?.find(saved => {
+        return saved.id === challenge.id;
+      });
+
+      // skip everything if a challenge is saved in db
```

Comment:
```suggestion
      // if the store is already populated with the saved files, we should not
      // overwrite them with the local storage data
```
It's clear from the code what's happening, but it was not obvious why we're doing that.